comictaggerlib/ui/talkeruigenerator.py

```diff
@@ -0,0 +1,213 @@
+        # int equals a spinbox (QSpinBox)
+        lbl = QtWidgets.QLabel(option.internal_name)
+        # Create a label
+        layout.addWidget(lbl, layout.rowCount() + 1, 0)
+        widget = QtWidgets.QSpinBox()
+        widget.setRange(0, 9999)
+        widget.setValue(value)
+        widget.setToolTip(option.help)
+        layout.addWidget(widget, layout.rowCount() - 1, 1, alignment=QtCore.Qt.AlignLeft)
+
+    if isinstance(value, float):
+        # float equals a spinbox (QDoubleSpinBox)
+        lbl = QtWidgets.QLabel(format_internal_name(option.internal_name))
+        # Create a label
+        layout.addWidget(lbl, layout.rowCount() + 1, 0)
+        widget = QtWidgets.QDoubleSpinBox()
+        widget.setRange(0, 9999.99)
+        widget.setValue(value)
+        widget.setToolTip(option.help)
+        layout.addWidget(widget, layout.rowCount() - 1, 1, alignment=QtCore.Qt.AlignLeft)
+
+    return layout, widget
+
+
+def generate_textbox(
+    option: settngs.Setting, value: str, layout: QtWidgets.QGridLayout
+) -> tuple[QtWidgets.QGridLayout, QtWidgets.QLineEdit, QtWidgets.QPushButton]:
+    btn = None
+    # str equals a text field (QLineEdit)
+    lbl = QtWidgets.QLabel(format_internal_name(option.internal_name))
+    # Create a label
+    layout.addWidget(lbl, layout.rowCount() + 1, 0)
+    widget = QtWidgets.QLineEdit()
+    widget.setObjectName(option.internal_name)
+    # Set widget status
+    widget.setText(value)
+    widget.setToolTip(option.help)
+    layout.addWidget(widget, layout.rowCount() - 1, 1)
+    # Special case for api_key, make a test button
+    if option.internal_name.endswith("api_key"):
+        btn = QtWidgets.QPushButton("Test Key")
+        layout.addWidget(btn, layout.rowCount() - 1, 2)
+
+    if option.internal_name.endswith("url"):
+        btn = QtWidgets.QPushButton("Test URL")
+        layout.addWidget(btn, layout.rowCount() - 1, 2)
+
+    return layout, widget, btn
+
+
+def generate_source_option_tabs(
+    parent: QtWidgets.QWidget,
+    tabs: QtWidgets.QTabWidget,
+    config: settngs.Config[settngs.Namespace],
+    talkers: dict[str, ComicTalker],
+) -> dict[str, QtWidgets.QWidget]:
+    """
+    Generate GUI tabs and settings for talkers
+    """
+
+    sources: dict = {}
+
+    # Add source sub tabs to Comic Sources tab
+    for talker_id, talker_obj in talkers.items():
+        # Add source to general tab dropdown list
+        tabs.findChildren(QtWidgets.QComboBox, "cobxInfoSource")[0].addItem(talker_obj.name, talker_id)
+
+        # Use a dict to make a var name from var
+        source_info = {}
+        tab_name = talker_id
+        source_info[tab_name] = {"tab": QtWidgets.QWidget(), "widgets": {}}
+        layout_grid = QtWidgets.QGridLayout()
+
+        for option in config[1][f"talker_{talker_id}"][1].values():
+            current_widget = None
+            if option.action is not None and (
+                isinstance(option.action, type(argparse.BooleanOptionalAction))
```

Comment:
All of these `isinstance` can be changed to `is` eg `option.action is argparse.BooleanOptionalAction`
And you should add `option.type is bool` to this one for completion